code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/Inbound.java

```diff
@@ -0,0 +1,111 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.messaging.internal;
+
+import java.util.Map;
+
+/**
+ * An {@link Inbound} object represents the common response (consequence) for AJO Campaigns targeting inbound channels.
+ */
+class Inbound {
+    // String representing a unique ID for this inbound item
+    private final String uniqueId;
+    // Enum representing the inbound item type
+    private final InboundType inboundType;
+    // Content for this inbound item e.g. inapp html string, or feed item JSON
+    private final String content;
+    // Contains mime type for this inbound item
+    private final String contentType;
+    // Represents when this inbound item went live. Represented in seconds since January 1, 1970
+    private final int publishedDate;
+    // Represents when this inbound item expires. Represented in seconds since January 1, 1970
+    private final int expiryDate;
+    // Contains additional key-value pairs associated with this inbound item
+    private final Map<String, Object> meta;
+
+    /**
+     * Constructor.
+     */
+    Inbound(final String uniqueId, final InboundType inboundType, final String content, final String contentType, final int publishedDate, final int expiryDate, final Map<String, Object> meta) {
+        this.uniqueId = uniqueId;
+        this.inboundType = inboundType;
+        this.content = content;
+        this.contentType = contentType;
+        this.expiryDate = expiryDate;
+        this.publishedDate = publishedDate;
+        this.meta = meta;
```

Comment:
Should we set empty values if string is null and 0 if published/ expiryDate < 0 for sanity.

Reply:
yes, i'll update